src/PotGenerator.php

```diff
@@ -35,12 +35,105 @@ public static function setCommentBeforeHeaders( $comment ) {
 	 * {@parentDoc}.
 	 */
 	public static function toString( Translations $translations, array $options = [] ) {
-		$result = '';
+		$lines   = static::$comments_before_headers;
+		$lines[] = 'msgid ""';
+		$lines[] = 'msgstr ""';
 
-		if ( ! empty( static::$comments_before_headers ) ) {
-			$result = implode( "\n", static::$comments_before_headers ) . "\n";
+		$plural_form = $translations->getPluralForms();
+		$plural_size = is_array( $plural_form ) ? ( $plural_form[0] - 1 ) : 1;
+
+		foreach ( $translations->getHeaders() as $name => $value ) {
+			$lines[] = sprintf( '"%s: %s\\n"', $name, $value );
+		}
+
+		$lines[] = '';
+
+		foreach ( $translations as $translation ) {
+			/** @var \Gettext\Translation $translation */
+			if ( $translation->hasComments() ) {
+				foreach ( $translation->getComments() as $comment ) {
+					$lines[] = '# ' . $comment;
+				}
+			}
+
+			if ( $translation->hasExtractedComments() ) {
+				foreach ( $translation->getExtractedComments() as $comment ) {
+					$lines[] = '#. ' . $comment;
+				}
+			}
+
+			foreach ( $translation->getReferences() as $reference ) {
+				$lines[] = '#: ' . $reference[0] . ( $reference[1] !== null ? ':' . $reference[1] : null );
```

Comment:
I'd say `''` is a better no-op here when concatenating:
```suggestion
				$lines[] = '#: ' . $reference[0] . ( $reference[1] !== null ? ':' . $reference[1] : '' );
```